File: backend/app/schemas/catalog/prompts.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class PromptTemplateUpsertRequest(BaseModel):
    name: str = Field(min_length=1)
    description: str = ""
    content: str = Field(min_length=1)


class PromptTemplateRevisionOut(ORMModel):
    id: str
    version: int
    content: str
    variables_json: list[str]
    checksum: str
    created_at: datetime


class PromptTemplateOut(ORMModel):
    id: str
    template_type: str
    name: str
    description: str
    active_revision_id: str | None
    created_at: datetime
    updated_at: datetime


class PromptTemplateDetail(PromptTemplateOut):
    active_revision: PromptTemplateRevisionOut | None = None

File: frontend/src/api/types/prompts.ts
export interface PromptTemplateRevisionRead {
  id: string;
  version: number;
  content: string;
  variables_json: string[];
  checksum: string;
  created_at: string;
}

export interface PromptTemplateRead {
  id: string;
  template_type: string;
  name: string;
  description: string;
  active_revision_id: string | null;
  created_at: string;
  updated_at: string;
  active_revision: PromptTemplateRevisionRead | null;
}

export interface PromptTemplatePayload {
  name: string;
  description: string;
  content: string;
}
